src/main/java/edu/kis/vh/nursery/FIFORyhmer.java

```diff
@@ -1,20 +1,28 @@
 package edu.kis.vh.nursery;
 
+import edu.kis.vh.nursery.stack.IntArrayStack;
+import edu.kis.vh.nursery.stack.IntStackInterface;
+
 public class FIFORyhmer extends DefaultCountingOutRyhmer {
 
-	private final DefaultCountingOutRyhmer temp = new DefaultCountingOutRyhmer();
+	public FIFORyhmer() {
+	}
+
+	public FIFORyhmer(IntStackInterface intLinkedList) {
+		super(intLinkedList);
+	}
+
+	private final IntStackInterface temp = new IntArrayStack();
 
 	@Override
 	public int countOut() {		
 		while (!callCheck())
-			temp.countIn(super.countOut());
-		int ret = temp.countOut();
-		while (!temp.callCheck())
-			countIn(temp.countOut());
+			temp.push(super.countOut());
+		int ret = temp.pop();
+		while (!temp.isEmpty())
+			countIn(temp.pop());
 		return ret;
 	}
-
-	public DefaultCountingOutRyhmer getTemp() {
-		return temp;
-	}
 }
+
+//IntArrayStack to lepszy wybór ponieważ w temp potrzebujemy jedynie ostatniej wartości więc rezerwujemy mniej pamięci

```

Comment:
No, temp pushes and later pops all values. If Ryhmer has more values than temps capacity the operation will fail. Implementation without limitations is needed (IntLinkedlist)